custom_components/google_home/models.py

```diff
@@ -3,25 +3,36 @@
 from __future__ import annotations
 
 from datetime import timedelta
-from typing import Any, Dict, List, Optional
+from typing import List, Optional, TypedDict
 
 from homeassistant.util.dt import as_local, utc_from_timestamp
 
-from .const import (
-    DATETIME_STR_FORMAT,
-    FIRE_TIME,
-    ID,
-    LABEL,
-    ORIGINAL_DURATION,
-    RECURRENCE,
-)
+from .const import DATETIME_STR_FORMAT
 
 
 def convert_from_ms_to_s(timestamp: int) -> int:
     """Converts from milliseconds to seconds"""
     return round(timestamp / 1000)
 
 
+class AlarmJsonDict(TypedDict, total=False):
+    """Typed dict for JSON representation of alarm returned by Google Home API"""
+
+    id: str
+    fire_time: int
+    label: Optional[str]
+    recurrence: Optional[str]
+
+
+class TimerJsonDict(TypedDict, total=False):
+    """Typed dict for JSON representation of timer returned by Google Home API"""
+
+    id: str
+    fire_time: int
+    original_duration: int
+    label: Optional[str]
```

Comment:
I suggest we move out `TypedDict` to a separate file like `types.py`. They are not "core" relevant, thou important to have. It would give it a bit better structure